js-extensions/README.md

```diff
@@ -1,4 +1,80 @@
-# ExtensionPoint npm module
+# Jenkins JavaScript Extensions
 
-This module is published via npm so that other plugins, external to the blueocean project, can make use of it. 
-Plugins can themselves make use of extension points. 
+Jenkins JavaScript Extensions are extensions which contribute to the UI in Jenkins BlueOcean.
+This module is used to define extension points - locations where the application accepts plugin-provided implementations.
+This module is also used to process the plugin extension point implementations to provide to BlueOcean.
+
+This module is published via npm as `@jenkins-cd/js-extensions` so that other plugins, external to the blueocean project, can make use of it.
+Plugins can themselves make use of extension points.
+
+Jenkins JavaScript Extensions are based on the extensibility model already established by Jenkins, based on data and views, with the ability to inherit views based on parent data types.
+
+Jenkins JavaScript Extensions: `@jenkins-cd/js-extensions` module exports 2 things:
+- `store` - the `ExtensionStore` instance (which must be initialized before it can be used)
+- `Renderer` - a React component to conveniently render extensions
+
+### Store API
+
+The `ExtensionStore` API is very simple, all public methods are asynchronous:
+
+- `getExtensions(extensionPointName, [filter,] onload)`
+    This method will async load data, filter the extensions based on the provided `filter`s, and call the onload handler with a list of extension exports, e.g. the React classes or otherwise exported references.
+    `filter` currently supports the following properties:
+    - `dataType` - to filter components that accept specific types of data from the server
+    - `componentType` to enforce a specific component type is returned (e.g. `Link`)
+
+- `getTypeInfo(type, onload)`
+    This will return a list of type information, from the [classes API](../blueocean-rest/README.md#classes_API), this method also handles caching results locally.
+
+- `init()`
+    Required to be called with `{ extensionDataProvider: ..., typeInfoProvider: }` see: [ExtensionStore.js](src/ExtensionStore.js#init) for details. This is currently done in [init.jsx](../blueocean-web/src/main/js/init.jsx), with methods to fetch extension data from `<jenkins-url>/blue/js-extensions/` and type information from `<jenkins-url>/blue/rest/classes/<class-name>`.
```

Comment:
I think it should be made more explicit (it is somewhat implicit in "This is currently done in ...", but not sure that helps a lot) that this method is not something that users of this API should ever call (? in fact, is it idempotent ?). I'd nearly not publicise it's existence at all.